src/base/name/relative.rs

```diff
@@ -136,6 +136,12 @@ impl RelativeDname<[u8]> {
     }
 
     /// Creates a relative domain name from an octet slice.
+    ///
```

Comment:
Since this indeed seems to be a recurring mistake, perhaps we should add a sentence on all of these that the slice needs to in wire format, i.e., no dots but each label prefixed with its length?